token/typed.go

```diff
@@ -0,0 +1,59 @@
+package token
```

Comment:
The issue closed itself, sorry. Re-opening here.
"NewTyped, ListTyped, RegisterTyped, typedLookup, CreateTypedFunc and ArgumentsTypedParser" sounds coherent.
However I'm still wondering if this shouldn't move to a new `token/typed` package.
- An almost empty package is not that bad if it permits a clean API;
- The functions are going to get mixed up with the whole token package in godoc and this will get worse when adding e.g. the token attributes.